packages/core/src/routes/profile/index.ts

```diff
@@ -61,7 +71,20 @@ export default function profileRoutes<T extends UserRouter>(
       const { id: userId, scopes } = ctx.auth;
       const { body } = ctx.guard;
       const { name, avatar, username } = body;
+      const { fields } = ctx.accountCenter;
 
+      assertThat(
+        name === undefined || fields.name === AccountCenterControlValue.Edit,
+        'account_center.filed_not_editable'
+      );
+      assertThat(
+        avatar === undefined || fields.avatar === AccountCenterControlValue.Edit,
+        'account_center.filed_not_editable'
+      );
+      assertThat(
+        username === undefined || fields.username === AccountCenterControlValue.Edit,
+        'account_center.filed_not_editable'
+      );
```

Comment:
use for each?

Reply:
I tried, but it is very hard to resolve typing issues. So I believe that won't be better